File: HCMUS-BigInt/benchmark/cplusplus/HelloWorld.cpp
//
// Created by Phuc Nguyen on 07/02/2024.
//


#include <iostream>
#include <chrono>

int main() {
    auto start_time = std::chrono::high_resolution_clock::now();

    for (int i = 0; i < 100000; i++) {
        std::cout << "Hello World";
    }

    auto end_time = std::chrono::high_resolution_clock::now();

    auto elapsed_time = std::chrono::duration_cast<std::chrono::microseconds>(end_time - start_time).count();
    std::cout << "\nTotal time elapsed: " << elapsed_time << " microseconds" << std::endl;

    return 0;
}

File: HCMUS-BigInt/benchmark/python/HelloWorld.py
import time

start_time = time.time()

for _ in range(100000):
    print("Hello World", end="")

end_time = time.time()

elapsed_time = (end_time - start_time) * 1000000  # Convert to microseconds
print(f"\nTotal time elapsed: {elapsed_time} microseconds")